XLS-30/README.md

```diff
@@ -0,0 +1,1254 @@
+|  `issuer`  | :heavy_check_mark: | specifies the unique XRPL account address of the entity issuing the currency                                    |
+| `currency` | :heavy_check_mark: | arbitrary code for currency to issue                                                                            |
+
+If the asset is XRP, then the issuer subfield is not mentioned.      
+
+---
+| Field Name |     Required?      | JSON Type | Internal Type |
+| ---------- | :----------------: | :-------: | :-----------: |
+| `Asset2` | :heavy_check_mark: | `object` |   `ISSUE`   |
+
+`Asset2` specifies the other asset of the AMM instance against which the transaction is to be executed.
+
+---
+| Field Name        | Required? |      JSON Type       | Internal Type |
+| ----------------- | :-------: | :------------------: | :-----------: |
+| `Amount` |           | `string` or `object` |   `AMOUNT`    |
+
+`Amount` specifies the amount of one of the pools assets. If the asset is XRP, then the `Amount` is a `string` specifying the number of drops. Otherwise it is an `object` with the following subfields:
+
+| Field name |     Required?      | Description                                                                                                     |
+| :--------: | :----------------: | :-------------------------------------------------------------------------------------------------------------- |
+|  `issuer`  | :heavy_check_mark: | specifies the unique XRPL account address of the entity issuing the currency                                    |
+| `currency` | :heavy_check_mark: | arbitrary code for currency to issue                                                                            |
+|  `value`   |                    | specifies the maximum amount of this currency, in decimal representation, that the trader is willing to add |
+
+---
+
+| Field Name       | Required? | JSON Type | Internal Type |
+| ---------------- | :-------: | :-------: | :-----------: |
+| `Amount2` |           | `string` or `object` |   `AMOUNT`    |
+
+`Amount2` specifies the details of other pool asset that the trader is willing to add.
+
+---
+
+| Field Name | Required? | JSON Type | Internal Type |
+| ---------- | :-------: | :-------: | :-----------: |
+| `EPrice`    |           | `string`  |   `AMOUNT`    |
+
+`EPrice` specifies the effective-price of the token out after successful execution of the transaction. For **`AMMDeposit`** transaction, the token out is always `LPToken`. 
+
+Note that the relative pricing does not change in case of **all-asset** deposit transaction.
+`EPrice` is an invalid field for all assets deposits. It should only be specified in case of single sided deposits.
+
+---
+
+| Field Name | Required? | JSON Type | Internal Type |
+| ---------- | :-------: | :-------: | :-----------: |
+| `LPTokenOut` |           | `string`  |   `AMOUNT`    |
+
+`LPTokenOut` specifies the amount of shares of the AMM instance pools.
+
```

Comment:
Could also have `TradingFee`, which is valid if the option flag is `tfTwoAssetIfEmpty`.

Reply:
Added.